### ISSUE TITLE
Adding org.jooq.Meta#getDefaultCatalog() & org.jooq.Catalog#getDefaultSchema()

### ISSUE THREAD

**Issue by rgoldberg** [jOOQ/jOOQ#4559]
It would be useful to add the following methods:

    Catalog org.jooq.Meta#getDefaultCatalog()
    Schema org.jooq.Meta#getDefaultSchema()
    Schema org.jooq.Catalog#getDefaultSchema()

So I can write:

    using(jooqConfig).meta().getDefaultSchema().getTable("table_name")

If there is no appropriate default, the methods could either throw an exception, or could return `null`.

**Comment by lukaseder** [comment]
Good idea. This would work nicely with #3783, which allows to select the `CURRENT_SCHEMA()` in queries.

**Comment by lukaseder** [comment]
Having another look at this idea. There are mainly two types of default schema (or catalog):

- Server side defaults (where "server" can also mean XML file)
  - JDBC `DatabaseMetaData` backed default
  - XML `InformationSchema` backed default (new)
  - DDL default
  - Code generation schema default
- Client side defaults
  - `Settings.interpreterSearchPath`
  - `Settings.parserSearchPath`
  - `RenderMapping.defaultSchema`, etc.

The question is how these would play with each other, and it does appear to be a rather complex question. The complexity certainly doesn't justify adding the API just for convenience. It's very easy to configure your own defaults somewhere (or use one of the above), and then just call `Meta::filterSchemas` or whatever.

So, I'll put this task on hold for now. (always good to not rush such feature implementations)
